lib/ClientConnectContext.php

```diff
@@ -52,6 +54,21 @@ public function getMaxAttempts(): int {
         return $this->maxAttempts;
     }
 
+    public function withResolveTypeRestriction(int $type): self {
```

Comment:
`$type` should probably allow `null` to remove the type restriction.

Reply:
Should probably be named `withIpTypeRestriction`?